File: pyLab2/FileEditing.py
import  FileIO
from datetime import datetime


def EditItemList(items):
    editedItems = []
    for item in items:
        data = item.split()
        prodDate = datetime.strptime(data[2], "%d.%m.%Y").date()
        expireDate = datetime.strptime(data[3], "%d.%m.%Y").date()
        prodExpireInterval = (expireDate - prodDate).days
        todayExpireInterval = (expireDate - (datetime.today()).date()).days

        if todayExpireInterval < prodExpireInterval/10:
            editedItems.append(item)
    return editedItems


def GetLastDaysItems(previousItems):
    lastDaysItems = []
    print("List of items that were produced within 10 days:")
    for item in previousItems:
        data = item.split()
        prodDate = datetime.strptime(data[2], "%d.%m.%Y").date()
        if (datetime.today().date() - prodDate).days <= 10:
            lastDaysItems.append(item)
    return lastDaysItems


def PrintList(items):
    for item in items:
        data = item.split()
        print(data[0] + ' ' + data[1] + ' ' + data[2] + ' ' + data[3])

File: Lab_2/Lab2/Lab2/FileEditing.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Lab2;

namespace Lab2
{
    internal class FileEditing
    {
        public static List<Item> EditItemList(List<Item> items)
        {
            double prodExpireInterval = 0;
            double todayExpireInterval = 0;
            DateTime prodDate = new DateTime();
            DateTime expireDate = new DateTime();
            List<Item> editedItems = new List<Item>();
            Item currentItem = new Item("", "", "", 0);

            for (int i = 0; i < items.Count; i++)
            {
                currentItem = items[i];               
                prodDate = Convert.ToDateTime(currentItem.GetProdDate());
                expireDate = Convert.ToDateTime(currentItem.GetExpireDate());
                prodExpireInterval = (expireDate.Subtract(prodDate).TotalDays);
                todayExpireInterval = (expireDate.Subtract(DateTime.Today).TotalDays);

                if(todayExpireInterval < prodExpireInterval/10)
                {
                    editedItems.Add(currentItem);
                }
            }

            return editedItems;
        }

        public static List<Item> GetLastDaysItems(List<Item> items)
        {
            List<Item> lastDaysItems = new List<Item>();
            Item currentItem = new Item("", "", "", 0);
            DateTime prodDate = new DateTime();

            for (int i = 0; i < items.Count; i++)
            {
                currentItem = items[i];
                prodDate = Convert.ToDateTime(currentItem.GetProdDate());
                if(((DateTime.Today).Subtract(prodDate)).TotalDays <= 10)
                {
                    lastDaysItems.Add(currentItem);
                }
            }

            return lastDaysItems;
        }

        public static void PrintList(List<Item> items)
        {
            for (int i = 0; i < items.Count; i++)
            {
                Console.WriteLine(items[i].GetAllAtributes());
            }
        }
    }
}
